### PR TITLE
feat: Add campaign

### PR DESCRIPTION
**Pull request by muhsin-k**
- Added modal to create new campaigns
- Setup campaign state (actions, mutations and getters)
- Integrated the `addCampaign` API 
- Fixed the modal style override issue
- https://github.com/chatwoot/chatwoot/pull/2173 Review fixes

**Preview**
Screenshot 2021-04-30 at 6 55 58 PM

### CODE REVIEW

**Comment by pranavrajs** [comment]
@nithindavid Please review the PR.

**Comment by pranavrajs** [comment]
@muhsin-k A note: Please add specs.

**Comment by muhsin-k** [comment]
@nithindavid Added the specs. Please review the PR

File: app/javascript/dashboard/assets/scss/widgets/_modal.scss [hunk 1]
```diff
@@ -30,7 +30,7 @@
 }
 
 
-.page-top-bar {
+.modal-top-bar {
```
Comment (nithindavid) [inline app/javascript/dashboard/assets/scss/widgets/_modal.scss:33]: There seems to be some conflict with renaming this class.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+            />
+            {{ $t('CAMPAIGN.ADD.FORM.ENABLED') }}
+          </label>
+        </div>
+
+        <div class="modal-footer">
+          <div class="medium-12 columns">
+            <woot-submit-button
+              :disabled="
+                $v.message.$invalid ||
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:97]: Can we move this condition to a computed since this has more than 3 variables? Which would improve readability

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+      </form>
+    </div>
+  </modal>
+</template>
+
+<script>
+import { mapGetters } from 'vuex';
+import { required, url, minLength } from 'vuelidate/lib/validators';
+
+import WootSubmitButton from '../../../../../components/buttons/FormSubmitButton';
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:121]: Can we use `<woot-button>` with `type="submit"` here ?( It's available without import, ie global )

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+import Modal from '../../../../../components/Modal';
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:122]: For common components can we use relative imports ?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+      uiFlags: 'campaigns/getUIFlags',
+    }),
+
+    agentsList() {
+      return this.agentList;
+    },
+  },
+  methods: {
+    addCampaign() {
+      this.$store
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:180]: We are mostly following async await for this syntax. Can we use the same if it's possible here ?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/Campaign.vue [hunk 6]
```diff
@@ -1,17 +1,17 @@
 <template>
   <div class="column content-box">
-    <div class="row">
-      <a class="button icon success nice button--fixed-right-top">
+    <div v-if="campaigns.length" class="row button-wrapper">
+      <a class="button icon success nice" @click="openAddPopup">
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/Campaign.vue:4]: Use woot-button since this isn't a link

Comment (nithindavid) [reply app/javascript/dashboard/routes/dashboard/settings/inbox/components/Campaign.vue:4]: Use variant="clear link"

**Review by nithindavid — CHANGES_REQUESTED** [review]
Please see the comments, let me know if there is any issue.

**Comment by muhsin-k** [comment]
@nithindavid Changes are updated as per review

**Review by nithindavid — APPROVED** [review]

File: app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+  computed: {
+    ...mapGetters({
+      agentList: 'agents/getAgents',
+      uiFlags: 'campaigns/getUIFlags',
+    }),
+
+    agentsList() {
+      return this.agentList;
+    },
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:166]: @muhsin-k Why is this required? Can't we rename the agentList defined in mapGetter?

Comment (muhsin-k) [reply app/javascript/dashboard/routes/dashboard/settings/inbox/components/AddCampaign.vue:166]: Yes. I will update this campaign table PR
